apps/backend/src/main/java/ch/heig/pdg/backend/services/InventoryService.java

```diff
@@ -27,6 +35,41 @@ public List<InventoryDTO> getInventories(HugoSearchFilter<Inventory> filter) {
                 .collect(Collectors.toList());
     }
 
+    public List<SearchResultDTO> searchInventory(String searchTerm, Integer inventoryId) {
+        Stream<SearchResultDTO> users = this.userService.search(inventoryId, searchTerm).stream().map(
+                u -> {
+                    SearchResultDTO dto = new SearchResultDTO();
+                    dto.id(u.getId());
+                    dto.entityType("user");
+                    dto.description("");
+                    dto.name(u.getFirstName() + " " + u.getLastName());
+                    return dto;
+                }
+        );
+        Stream<SearchResultDTO> locations = this.locationService.search(inventoryId, searchTerm).stream().map(
+                l -> {
+                    SearchResultDTO dto = new SearchResultDTO();
+                    dto.id(l.getId());
+                    dto.entityType("location");
+                    dto.description(l.getDescription());
+                    dto.name(l.getName());
+                    return dto;
+                }
+        );
+        Stream<SearchResultDTO> itemModels = this.itemModelService.search(inventoryId, searchTerm).stream().map(
+                i -> {
+                    SearchResultDTO dto = new SearchResultDTO();
+                    dto.id(i.getId());
+                    dto.entityType("itemModel");
+                    dto.description(i.getDescription());
+                    dto.name(i.getName());
+                    return dto;
+                }
+        );
+
+        return Stream.of(locations, users, itemModels).flatMap(o -> o).collect(Collectors.toList());
```

Comment:
Is there a reason to `flatMap` here?
Would'nt it be easier to return a dict of the results (`{"locations": [], "items-models": [], "users": []}`) and avoid the transformations above? 

> No special need to change

Reply:
It's to simplify the declaration in the api spec and to streamline the data, but I'm happy to change it if required.